src/components/modals/ant-management/ReassignNameModal/ReassignNameModal.tsx

```diff
@@ -0,0 +1,643 @@
+          antId: newAntProcessId,
+        }
+      : {},
+  );
+
+  const [antType, setAntType] = useState<
+    | REASSIGN_NAME_WORKFLOWS.NEW_BLANK
+    | REASSIGN_NAME_WORKFLOWS.NEW_EXISTING
+    | REASSIGN_NAME_WORKFLOWS.EXISTING
+  >(REASSIGN_NAME_WORKFLOWS.NEW_BLANK);
+  const [accepted, setAccepted] = useState<boolean>(false);
+
+  function handleClose() {
+    setWorkflow(undefined);
+    setNewAntProcessId('');
+    setShow(false);
+  }
+
+  async function handleReassign() {
+    try {
+      if (!wallet?.contractSigner) {
+        throw new Error('No ArConnect Signer found');
+      }
+      if (!walletAddress)
+        throw new Error('Must connect to Reassign the domain');
+
+      const previousState: SpawnANTState = {
+        controllers: domainData.controllers,
+        records: domainData.records,
+        owner: walletAddress.toString(),
+        ticker: domainData.info.Ticker,
+        name: domainData.info.Name,
+        description: domainData.info.Description,
+        keywords: domainData.info.Keywords,
+        balances: domainData.state.Balances,
+      };
+
+      const result = await dispatchANTInteraction({
+        signer: wallet.contractSigner,
+        payload: {
+          name,
+          arioProcessId,
+          // if we provide newAntProcessId it will skip spawning the ant and ignore previousState
+          newAntProcessId:
+            antType === REASSIGN_NAME_WORKFLOWS.EXISTING
+              ? newAntProcessId
+              : undefined,
+          previousState:
+            antType === REASSIGN_NAME_WORKFLOWS.NEW_BLANK
+              ? undefined
+              : previousState,
+          luaCodeTxId: DEFAULT_ANT_LUA_ID,
+        },
+        processId,
+        workflowName: ANT_INTERACTION_TYPES.REASSIGN_NAME,
+        dispatch: dispatchTransactionState,
+        owner: walletAddress.toString(),
+        ao: aoClient,
+      });
+      eventEmitter.emit('success', {
+        message: (
+          <div>
+            <span>
+              Reassign Name completed with transaction ID:{' '}
+              <ArweaveID
+                characterCount={8}
+                shouldLink={true}
+                type={ArweaveIdTypes.INTERACTION}
+                id={new ArweaveTransactionID(result.id)}
+              />
+            </span>
+          </div>
+        ),
+        name: ANT_INTERACTION_TYPES.REASSIGN_NAME,
+      });
+      queryClient.resetQueries({
+        queryKey: ['handlers', processId],
+      });
+      queryClient.resetQueries({
+        queryKey: ['domainInfo', name],
+      });
+      queryClient.resetQueries({
+        queryKey: ['domainInfo', processId],
+      });
+
+      dispatchArNSUpdate({
+        walletAddress: walletAddress,
+        arioProcessId,
+        dispatch: dispatchArNSState,
+        emitter: arnsEmitter,
+        ao: aoClient,
+      });
+      handleClose();
+    } catch (error) {
+      eventEmitter.emit('error', error);
+    }
+  }
+
+  if (!show) return <></>;
```

Comment:
nit: could have this at the top of the component to prevent the above code from running

Reply:
Hooks need to be mounted deterministically

Reply:
That being said, because this uses the useDomainInfo hook, it *shouldnt* be doing anything extra here, since it is being access from a page that is also using that hook - internally it uses useQuery which will dedupe the request and return the same data to all the components calling it.